Assets/00_Content/Scripts/Interactables/Beers/BeerTap.cs

```diff
@@ -14,12 +38,43 @@ public class BeerTap : Interactable {
 				return;
 			}
 
-			Debug.Log("Pouring beer...");
-			Instantiate(beerPrefab, beerSpawnpoint.position, Quaternion.identity);
+			if (itemSlot.HasItemInSlot) {
+				Debug.Log("Can�t pour, has full beer in slot");
+				return;
+			}
+
+			StartCoroutine(PourBeer());
 		}
 
 		public override void CancelInteraction() {
 			base.CancelInteraction();
 		}
+
+		private IEnumerator PourBeer() {
+
+			while (!itemSlot.HasItemInSlot && buttonInteract.isPressed && pouringProgress <= 100) {
+
+				
+				pouringProgress += 1 * pourTimeMultiplier; 
+
+				if(!progressBar.activeInHierarchy)
+					progressBar.SetActive(true);
+
+				progressBarImage.fillAmount = pouringProgress * 0.01f;
+
+				if(pouringProgress > 100) {
```

Comment:
Potential edge case: The surrounding while loop only allows for progress up to and including 100. So if the progress lands at exactly 100, this code will not run.

Reply:
if it lands on 100, it will still redo the while-loop and increase the value to go above.

Reply:
Ah, true. My bad!